xbmc/guilib/GUIVisualisationControl.cpp

```diff
@@ -242,7 +242,7 @@ void CGUIVisualisationControl::OnInitialize(int channels, int samplesPerSec, int
 
 void CGUIVisualisationControl::OnAudioData(const float* audioData, unsigned int audioDataLength)
 {
-  if (!m_instance || !m_alreadyStarted)
+  if (!m_instance || !m_alreadyStarted || !audioData || !audioDataLength)
```

Comment:
Please use "audioDataLength == 0". It's more clear to the reader it's not a pointer.

Reply:
Absolutely. Change made. Thanks!